packages/inventory/src/Stocktake/DetailView/columns.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  GenericColumnKey,
+  useColumns,
+  ColumnAlign,
+  SortBy,
+  SortRule,
+} from '@openmsupply-client/common';
+import { StocktakeItem } from './../../types';
+
+interface UseStocktakeColumnOptions {
+  sortBy: SortBy<StocktakeItem>;
+  onChangeSortBy: (
+    newSortRule: SortRule<StocktakeItem>
+  ) => SortBy<StocktakeItem>;
+}
+
+export const useStocktakeColumns = ({
+  sortBy,
+  onChangeSortBy,
+}: UseStocktakeColumnOptions) =>
+  useColumns<StocktakeItem>(
+    [
+      'itemCode',
+      'itemName',
+      'batch',
+      'expiryDate',
```

Comment:
I've had a wee play and added an expiry date formatter to `ColumnDefinitionSetBuilder`. what do you say to something like this?

```
  expiryDate: {
    key: 'expiryDate',
    label: 'label.expiry',
    width: 200,
    format: ColumnFormat.Date,
    formatter: dateString =>
      dateString ? formatExpiryDate(new Date(dateString as string)) || '' : '',
  },
```

Reply:
Yup great idea - been avoiding the formatters a little as they're not type safe :(